feat: add Developer Experience list

prokopsimek opened the pull request:
<!-- Congrats on creating an Awesome list! 🎉 -->

<!-- Please fill in the below placeholders -->

**https://github.com/prokopsimek/awesome-developer-experience/**

**The developer experience is like UX but more oriented to development. The list contains resources, tools, and tips to improve the dev team's effectiveness and better API/Docs delivery to developers.**

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

I have reviewed 3 PRs earlier. I hope my comments are enough to support being the awesome project self-sustainable.
- https://github.com/sindresorhus/awesome/pull/2007#issuecomment-835217620
- https://github.com/sindresorhus/awesome/pull/1954#issuecomment-835220660
- https://github.com/sindresorhus/awesome/pull/1950#issuecomment-835221917

I have reviewed 3 **new** PRs. I hope my comments are enough to support being the awesome project self-sustainable.
- https://github.com/sindresorhus/awesome/pull/2270#issuecomment-1162197413
- https://github.com/sindresorhus/awesome/pull/2278#issuecomment-1162193420
- https://github.com/sindresorhus/awesome/pull/2289#issuecomment-1162285544

`awesome-lint` is satisfied (✔ Linting)

Thank you for your reviews! I am here to improve whatever could be improved. I also know that the list is not long enough as other lists but I believe in support from the community to improve the list. 

Please, suggest any changes or improvements.

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. **Just commenting “looks good” or simply marking the pull request as approved does not count!** You have to actually point out mistakes or improvement suggestions.
- You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/main/create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `add Swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The title of your entry should be title-cased and the URL to your list should end in `#readme`.
	- Example: `- [Software Architecture](https://github.com/simskij/awesome-software-architecture#readme) - The discipline of designing and building software.`
- The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it. **Instead use [#2242](https://github.com/sindresorhus/awesome/issues/2242) for incubation visibility**.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/main/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
	- Must not feature `Contributing` or `Footnotes` sections.
- Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=main&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name, text, or a `Licence` section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/main/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
	- It can optionally be linked from the readme in a dedicated section titled `Contributing`, positioned at the top or bottom of the main content.
	- The section should not appear in the Table of Contents.
- All non-important but necessary content (like extra copyright notices, hyperlinks to sources, pointers to expansive content, etc) should be grouped in a `Footnotes` section at the bottom of the readme. The section should not be present in the Table of Contents.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

sindresorhus commented:
You haven't even addressed the feedback I gave. You also never responded to my feedback then.

sindresorhus commented:
It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

moh01986 commented:
I do not uderstand any things at github
Need help

prokopsimek commented:
@sindresorhus Hi, we added many new links about DX so I believe that this https://github.com/sindresorhus/awesome/pull/2010#issuecomment-887769444 comment is resolved with new content including this https://github.com/sindresorhus/awesome/pull/2010#issuecomment-887771379. What do you think? Should I do better?

Sorry, I forgot the note about the description (https://github.com/sindresorhus/awesome/pull/2010#issuecomment-887770372). I changed it to explain it better.

Thanks for your review and guidance.

notpresident35 commented:
- First impression says Resources should come first. Especially if you're putting "What is DX" there, it should be as close to the top of the page as possible for new users. 
- Resources description also cuts off with a hanging "and"
- The description of DX should come after the one-line description of your list. Really, I'm not sure if you need to include it - just redirect the users to somewhere that explains it. Also, if you got that description from somewhere, please cite it.
- Descriptions after each sub-category feel like filler, remove them and offer any lost information in the descriptions for each individual link, as well as resources that clearly explain important terminology.

prokopsimek commented:
@notpresident35 Thank you for the review! Can @sindresorhus confirm that these suggestions are a way of merging the list into a list of lists? Thanks!

Lissy93 reviewed (commented):
I think Sindre's [original comment](https://github.com/sindresorhus/awesome/pull/2010#issuecomment-887771379) is still valid. You've put your own tools at the top of each list. Maybe sort them in alphabetical order, or by another less biased metric?

For reference, I'm speaking about [dxscanner.io](https://dxscanner.io), [What is DX](https://developerexperience.io/practices/good-developer-experience), [developerexperiencemanifesto.org](https://developerexperiencemanifesto.org), etc.
Not sure which of the other services your associated with, but keeping things impartial should be important.

notpresident35 reviewed (commented):
(reviewed earlier, see above comment)

prokopsimek commented:
@Lissy93 @notpresident35 Thank you for your review. 

- I have ordered all items in lists alphabetically.
- I moved resources in the first place. I was thinking if documentation (alphabetically before resources) should be first, but I think it's better first to give some resources to learn about the topic and tools and more details right after that.
- I removed 2 links for DX Scanner as it's deprecated and unnecessary.

cosydney commented:
[Raycast](https://www.raycast.com/) is only for mac users but definitely a go to tool for any developers on a Mac ! I use it multiple times per day and saves me hours of my time.

cosydney reviewed (changes requested):
When I run the linter I get this minor typo error: 

  ✖  70:4  Text "Youtube" should be written as "YouTube"  remark-lint:awesome-spell-check

silopolis commented:
I believe this would find a good place in the proposed Technical Documentation category proposed in #2486.

sindresorhus commented:
I have decided to pass on this list. This pull request is still not fully compliant and I don't want to spend more time reviewing this.
